src/Lecture2.hs

```diff
@@ -155,12 +178,70 @@ You're free to define any helper functions.
 
 -- some help in the beginning ;)
 data Knight = Knight
-    { knightHealth    :: Int
-    , knightAttack    :: Int
-    , knightEndurance :: Int
-    }
-
-dragonFight = error "TODO"
+  { knightHealth    :: Int
+  , knightAttack    :: Int
+  , knightEndurance :: Int
+  }
+
+knightTurn :: (Knight, Dragon) -> (Knight, Dragon)
+knightTurn (knight, dragon) =
+  (knight {knightEndurance = (knightEndurance knight) - 1}
+  ,dragon {dragonHealth = (dragonHealth dragon) - (knightAttack knight)})
+
+data Treasure = Treasure
+  deriving (Show, Eq)
+data Chest = Chest
+  { chestGold     :: Natural
+  , chestTreasure :: Maybe Treasure
+  }
+  deriving (Show, Eq)
+
+data DragonType = Red | Black | Green
+  deriving (Show, Eq)
+
+getRandomGoldAmount :: Natural
+getRandomGoldAmount = 4 -- todo, randomize
```

Comment:
You don't need to randomize it on the output, you can store the gold inside the dragon